src/services/transaction.service.ts

```diff
@@ -0,0 +1,97 @@
+import { translations } from '~app/common/config';
+import { executeAfterEventV2 } from '~root/services/events.service';
+import { getEventByTxHash } from '~root/services/contractEvent.service';
+import { delay } from '~root/services/utils.service';
+import { getValidator } from '~root/services/validator.service';
+
+export enum TransactionMethod {
+  RegisterValidator = 'RegisterValidator',
+  BulkRegisterValidator = 'BulkRegisterValidator',
+  RemoveValidator = 'RemoveValidator',
+  BulkRemoveValidator = 'BulkRemoveValidator',
+  ExitValidator = 'ExitValidator',
+  BulkExitValidator = 'BulkExitValidator',
+  ReactivateCluster = 'ReactivateCluster',
+  UpdateValidator = 'UpdateValidator',
+}
+
+const contractMethods = {
+  [TransactionMethod.RegisterValidator]: async (contract: Contract, payload: any) => await contract.registerValidator(...payload.values()),
+  [TransactionMethod.BulkRegisterValidator]: async (contract: Contract, payload: any) => await contract.bulkRegisterValidator(...payload.values()),
+  [TransactionMethod.RemoveValidator]: async (contract: Contract, payload: any) => await contract.removeValidator(...Object.values(payload)),
+  [TransactionMethod.BulkRemoveValidator]: async (contract: Contract, payload: any) => await contract.bulkRemoveValidator(...Object.values(payload)),
+  [TransactionMethod.ExitValidator]: async (contract: Contract, payload: any) => await contract.exitValidator(...Object.values(payload)),
+  [TransactionMethod.BulkExitValidator]: async (contract: Contract, payload: any) => await contract.bulkExitValidator(...Object.values(payload)),
+  [TransactionMethod.ReactivateCluster]: async (contract: Contract, payload: any) => await contract.reactivate(...Object.values(payload)),
+  [TransactionMethod.UpdateValidator]: async (contract: Contract, payload: any) => await contract.reactivate(...Object.values(payload)),
+};
+
+const finishTransactionExecutor = {
+  [TransactionMethod.RegisterValidator]: async (hash: string) => await executeAfterEventV2(async () => !!await getEventByTxHash(hash), delay),
+  [TransactionMethod.BulkRegisterValidator]: async (hash: string) => await executeAfterEventV2(async () => !!await getEventByTxHash(hash), delay),
+  [TransactionMethod.RemoveValidator]: async (publicKey: string) => await executeAfterEventV2(async () => !await getValidator(publicKey, true), delay),
+  [TransactionMethod.BulkRemoveValidator]: async (publicKey: string) => await executeAfterEventV2(async () => !await getValidator(publicKey, true), delay),
+  [TransactionMethod.ReactivateCluster]: async (hash: string) => await executeAfterEventV2(async () => !!await getEventByTxHash(hash), delay),
+  [TransactionMethod.UpdateValidator]: async (hash: string) => await executeAfterEventV2(async () => !!await getEventByTxHash(hash), delay),
+};
+
+export const transactionExecutor = async ({ methodType, contract, payload, isContractWallet, callbackAfterExecution }: {
```

Comment:
send payload as array, ...Object.keys is expensive operation

Reply:
+